src/Archives/AdaptiveHuffmanTree.cpp

```diff
@@ -146,14 +122,22 @@ namespace Archives
 
 
 
+	// Raise exception if nodeIndex is out of range
+	void AdaptiveHuffmanTree::VerifyValidNodeIndex(NodeIndex nodeIndex)
+	{
+		// Check that the nodeIndex is in range
+		if (nodeIndex >= m_NumNodes)
+		{
+			throw std::runtime_error("Index out of range");
```

Comment:
I like moving the repeated check and error throw into a separate function. If this error is raised by OP2Archive, the message would be fairly vague. Maybe we could include some more detail to help the user out. Something like "During data compression or decompression, an Adaptive Huffman Tree encountered an out of range Node Index"?